keps/sig-node/3857-rro-mounts/README.md

```diff
@@ -0,0 +1,1075 @@
+ 	// This field is beta in 1.10.
++	// When RecursiveReadOnly is set to Enabled, MountPropagation must be None or nil (which defaults to None).
+ 	// +optional
+ 	MountPropagation *MountPropagationMode
+ 	// Expanded path within the volume from which the container's volume should be mounted.
+@@ -1905,6 +1923,18 @@ const (
+ 	MountPropagationBidirectional MountPropagationMode = "Bidirectional"
+ )
+ 
++// RecursiveReadOnlyMode describes recursive-readonly mode.
++type RecursiveReadOnlyMode string
++
++const (
++	// RecursiveReadOnlyDisabled disables recursive-readonly mode.
++	RecursiveReadOnlyDisabled RecursiveReadOnlyMode = "Disabled"
++	// RecursiveReadOnlyIfPossible enables recursive-readonly mode if possible.
++	RecursiveReadOnlyIfPossible RecursiveReadOnlyMode = "IfPossible"
++	// RecursiveReadOnlyEnabled enables recursive-readonly mode, or raise an error.
++	RecursiveReadOnlyEnabled RecursiveReadOnlyMode = "Enabled"
++)
++
+ // VolumeDevice describes a mapping of a raw block device within a container.
+ type VolumeDevice struct {
+ 	// name must match the name of a persistentVolumeClaim in the pod
+@@ -2505,6 +2535,10 @@ type ContainerStatus struct {
+ 	// +featureGate=InPlacePodVerticalScaling
+ 	// +optional
+ 	Resources *ResourceRequirements
++	// Status of volume mounts.
++	// +listType=atomic
++	// +optional
++	VolumeMounts []VolumeMountStatus
+ }
+ 
+ // PodPhase is a label for the condition of a pod at the current time.
+@@ -2584,6 +2618,24 @@ const (
+ 	PodResizeStatusInfeasible PodResizeStatus = "Infeasible"
+ )
+ 
++// VolumeMountStatus shows status of volume mounts.
++type VolumeMountStatus struct {
++	// Corresponds to the name of the original VolumeMount.
++	Name string
++	// Corresponds to the mount path of the original VolumeMount.
++	MountPath string
++	// ReadOnly must be set to true or false.
++	// A nil value in the original VolumeMount must be translated to true or false,
++	// depending on the mount result.
++	ReadOnly bool
++	// RecursiveReadOnly must be set to Disabled, Enabled, or nil (for non-readonly mounts).
++	// An IfPossible value in the original VolumeMount must be translated to Disabled or Enabled,
++	// depending on the mount result.
++	// +featureGate=RecursiveReadOnlyMounts
++	// +optional
++	RecursiveReadOnly *RecursiveReadOnlyMode
++}
++
+ // RestartPolicy describes how the container should be restarted.
+ // Only one of the following restart policies may be specified.
+ // If none of the following policies is specified, the default one
+@@ -4446,6 +4498,17 @@ type NodeDaemonEndpoints struct {
+ 	KubeletEndpoint DaemonEndpoint
+ }
+ 
++// RuntimeClassInfo is a set of runtime class information.
++type RuntimeClassInfo struct {
++	// Runtime class name.
++	// Empty for the default runtime class.
++	// +optional
++	Name string
++	// RecursiveReadOnlyMounts is set to true if the runtime class supports RecursiveReadOnlyMounts.
++	// +optional
++	RecursiveReadOnlyMounts *bool
```

Comment:
Why do we need this if we have `IfPossible`?

Reply:
So that the feature availability can be shown in `k get nodes`